src/Traits/ValidationAwareTrait.php

```diff
@@ -17,22 +17,23 @@
  */
 trait ValidationAwareTrait
 {
-    protected function handleResponseException(ResponseInterface $response): void
+    protected function dispatchValidationException(InvalidStatusCodeException $exception): InvalidStatusCodeException|ValidationErrorsException
     {
+        $response = $exception->getResponse();
         if ($response->getStatusCode() === 400) {
```

Comment:
newline here please, also no need to wrap `return $exception;` with else statement